lntest/itest/lnd_multi-hop-payments.go

```diff
@@ -280,10 +314,94 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
 		}
 	}
 
+	// We expect Carol to have successful forwards and settles for
+	// her sends.
+	assertHtlcEvents(
+		t, numPayments, 0, numPayments, routerrpc.HtlcEvent_SEND,
+		carolEvents,
+	)
+
+	// Dave and Alice should both have forwards and settles for
+	// their role as forwarding nodes.
+	assertHtlcEvents(
+		t, numPayments, 0, numPayments, routerrpc.HtlcEvent_FORWARD,
+		daveEvents,
+	)
+	assertHtlcEvents(
+		t, numPayments, 0, numPayments, routerrpc.HtlcEvent_FORWARD,
+		aliceEvents,
+	)
+
+	// Bob should only have settle events for his receives.
+	assertHtlcEvents(
+		t, 0, 0, numPayments, routerrpc.HtlcEvent_RECEIVE, bobEvents,
+	)
+
 	ctxt, _ = context.WithTimeout(ctxb, channelCloseTimeout)
 	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPointAlice, false)
 	ctxt, _ = context.WithTimeout(ctxb, channelCloseTimeout)
 	closeChannelAndAssert(ctxt, t, net, dave, chanPointDave, false)
 	ctxt, _ = context.WithTimeout(ctxb, channelCloseTimeout)
 	closeChannelAndAssert(ctxt, t, net, carol, chanPointCarol, false)
 }
+
+// assertHtlcEvents consumes events from a client and ensures that they are of
+// the expected type and contain the expected number of forwards, forward
+// failures and settles.
+func assertHtlcEvents(t *harnessTest, fwdCount, fwdFailCount, settleCount int,
+	userType routerrpc.HtlcEvent_EventType,
```

Comment:
For test readability, a struct containing `fwdCount`, `fwdFailCount`, `settleCount` and `userType` would have been nice (non blocking)